### DISCUSSION TITLE
Ability to listen to recording everywhere there is a spectrograph

### DISCUSSION THREAD

**Discussion by phillipredshirt** [mcguirepr89/BirdNET-Pi#363] [Ideas]
When training my father in law on his new birdnetpi installation we noticed that on some pages you have the ability to listen to the recording of the spectrograph and on others you don't.

It would be helpful for the user if every spectrograph also had a play button.

**Comment by mcguirepr89** [comment]
Hi, @phillipredshirt 

Would you mind providing an example of a spectrogram for which you are not able to listen to its corresponding sound file?

Update: I think I understand, you mean in the "Recordings" page how clicking the spectrogram takes you to the sound file instead of being a video element with the spectogram as the background?

Like this page https://virginia.birdnetpi.com/views.php?view=Recordings&species=American+Crow

versus this page https://virginia.birdnetpi.com/views.php?view=Today%27s+Detections

**Reply by ehpersonal38** [reply]
The "Today's Detections" implementation of the mp3 w/ spectrogram image is obviously much nicer, but it also has the drawback that _every single_ spectrogram image has to be loaded on page load. i.e. you can't lazyload video posters, which need to all be loaded for you to be able to play any mp3s. (How is this still the behavior of modern browsers? 😠)

So to implement that across all the pages we'll need to implement "Load 40 More" into said pages as well.

**Reply by mcguirepr89** [reply]
I have been putting that off but can say I'll definitely get this completed next week. Ethan, you can't make such cool features and not expect people to want them everywhere!!!! :) Such a great feature ~~

Anywho, now I have the luxury of copying and pasting  :sunglasses: , which makes me _feel_ like I have something to do with it :sunglasses:

**Reply by phillipredshirt** [reply]
I'm not sure how useful spectrograms are to most users. The three new users I trained yesterday(all experienced birders) found the spectrograms to be a curiosity for a few seconds but were really more interested in listening to the mp3 and knowing the time, the total observed and the species.  

Edit: Oh and live sound feed was the other most requested item.

**Reply by ehpersonal38** [reply]
@mcguirepr89 Hehe, great!! If you do need any help let me know! Also if you want to copy and paste I would do it from an older commit like here: https://github.com/mcguirepr89/BirdNET-Pi/commit/0e21b005846a4d2a26688783cc7c993eed23a67f, it doesn't have all the stuff you don't need like flickr, legacy mode, etc.

@phillipredshirt I think at first the spectrograms may be less-than-useful because you don't have enough knowledge of what they "should" look like. But as you use the system more, you start to recognize a typical spectrogram for any given species in your area - which makes finding the interesting sound bytes MUCH easier. Of course, different strokes for different folks, and others may be more data-oriented than I. For that I would recommend using "Legacy Mode" on Today's Detections, which cuts out all the fluff and presents what you're after.

The live sound feed is actually implemented in two places, in the top right as "Live Audio" and as a separate live "Spectrogram" page. You'll find it to the right of "Today's Detections"

**Reply by mcguirepr89** [reply]
There is a live sound feed in the top right where you can click "Live Audio"

There is also live audio built into the "Spectrogram" view

It is my understanding that ornithologists highly value spectrogram data. That input is how its inclusion into BirdNET-Pi first arose, but I'm no ornithologist. I do, however, trust @DD4WH whose many years of experience in this field has informed that trust :) shout out :)

@ehpersonal38 is totally right, too, that your eye becomes an asset in (coincidentally) doing exactly what BirdNET is doing!! That is to say, uses visual data to confirm the audio is of a particular species. It will take little time for you to recognize that some (most) species have a truly unique signature to their spectrogram. I'm hopeful that even the most experienced birders will come to realize what a powerful tool it is to make connections with this new (in type and quantity of) data.

Also on that note, the "Species Stats" pages become _very_ rich when populated with longterm data as is so fun to peruse [on @CaiusX's installation](https://joburg.birdnetpi.com)

**Reply by phillipredshirt** [reply]
@ehpersonal38 thanks for the feedback. I'll keep that in mind and mention it to my FIL. I think the spectrogram is probably also useful for finding cleaner audio clips.

I didn't mean to suggest that we couldn't find the live audio button, just that it was one of the most requested features while we were going through it.  After playing with Merlin I think they were expecting instant feedback but after explaining how it works they now understand that the analysis is going to lag behind the live feed.

It will be interesting to see what they use the most after two weeks compared to what they were focused on during hour 1.

**Reply by mcguirepr89** [reply]
>It will be interesting to see what they use the most after two weeks compared to what they were focused on during hour 1.

Please do update us on that! It is super valuable feedback -- I think we get a lot of great feedback from users right after installation and maybe not as much from folks after sustained use? Hard to say, but please let us know if you can!

**Comment by Svardsten53** [comment]
I agree with **@phillipredshirt**. When you open a file in "Recordings" a new page is opended with just a playback control where you are not able to follow the sound on the spectrogram.
